lib/fileio/commands/baseImportCommand.cpp

```diff
@@ -0,0 +1,262 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "baseImportCommand.h"
+
+#include <pxr/pxr.h>
+#include <pxr/usd/ar/resolver.h>
+
+#include <maya/MArgList.h>
+#include <maya/MStatus.h>
+#include <maya/MSyntax.h>
+#include <maya/MSelectionList.h>
+
+#include <mayaUsd/fileio/jobs/jobArgs.h>
+#include <mayaUsd/fileio/jobs/readJob.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+MAYAUSD_NS_DEF {
+
+MayaUSDImportCommand::MayaUSDImportCommand()
+{
+}
+
+/* virtual */
+MayaUSDImportCommand::~MayaUSDImportCommand()
+{
+}
+
+/* static */
+MSyntax
+MayaUSDImportCommand::createSyntax()
+{
+    MSyntax syntax;
+
+    // These flags correspond to entries in
+    // UsdMayaJobImportArgs::GetDefaultDictionary.
+    syntax.addFlag(kShadingModeFlag,
+                   UsdMayaJobImportArgsTokens->shadingMode.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kAssemblyRepFlag,
+                   UsdMayaJobImportArgsTokens->assemblyRep.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kMetadataFlag,
+                   UsdMayaJobImportArgsTokens->metadata.GetText(),
+                   MSyntax::kString);
+    syntax.makeFlagMultiUse(kMetadataFlag);
+    syntax.addFlag(kApiSchemaFlag,
+                   UsdMayaJobImportArgsTokens->apiSchema.GetText(),
+                   MSyntax::kString);
+    syntax.makeFlagMultiUse(kApiSchemaFlag);
+    syntax.addFlag(kExcludePrimvarFlag,
+                   UsdMayaJobImportArgsTokens->excludePrimvar.GetText(),
+                   MSyntax::kString);
+    syntax.makeFlagMultiUse(kExcludePrimvarFlag);
+    syntax.addFlag(kUseAsAnimationCacheFlag,
+                   UsdMayaJobImportArgsTokens->useAsAnimationCache.GetText(),
+                   MSyntax::kBoolean);
+
+    // These are additional flags under our control.
+    syntax.addFlag(kFileFlag, kFileFlagLong, MSyntax::kString);
+    syntax.addFlag(kParentFlag, kParentFlagLong, MSyntax::kString);
+    syntax.addFlag(kReadAnimDataFlag, kReadAnimDataFlagLong, MSyntax::kBoolean);
+    syntax.addFlag(kFrameRangeFlag, kFrameRangeFlagLong, MSyntax::kDouble, MSyntax::kDouble);
+    syntax.addFlag(kPrimPathFlag, kPrimPathFlagLong, MSyntax::kString);
+    syntax.addFlag(kVariantFlag, kVariantFlagLong, MSyntax::kString, MSyntax::kString);
+    syntax.makeFlagMultiUse(kVariantFlag);
+
+    syntax.addFlag(kVerboseFlag, kVerboseFlagLong, MSyntax::kNoArg);
+
+    syntax.enableQuery(false);
+    syntax.enableEdit(false);
+
+    return syntax;
+}
+
+/* static */
+void*
+MayaUSDImportCommand::creator()
+{
+    return new MayaUSDImportCommand();
+}
+
+/* virtual */
+std::unique_ptr<UsdMaya_ReadJob> MayaUSDImportCommand::initializeReadJob(const MayaUsd::ImportData & data, 
+        const UsdMayaJobImportArgs & args)
+{
+    return std::unique_ptr<UsdMaya_ReadJob>(new UsdMaya_ReadJob(data, args));

```

Comment:
This is the one place that the Pixar plugin will do something different.  It will override this and return a pointer to a different type of Job, the MayaUSD command will just use the one the base class creates.